todo_app/app.py

```diff
@@ -1,15 +1,64 @@
-from flask import Flask
-
+import sys
+from flask import Flask, render_template, request, redirect
 from todo_app.flask_config import Config
+from todo_app.data.session_items import *
 
 app = Flask(__name__)
 app.config.from_object(Config)
 
+#error handling for 404
+@app.errorhandler(404)
+def not_found(e):
+    return render_template("404.html")
 
-@app.route('/')
+#default
+@app.route('/', methods=['GET', 'POST'])
 def index():
-    return 'Hello World!'
+    #return 'Hello World!'
+    if request.method == 'POST':
+        task_title = request.form['title']
+        task_status = request.form['status']
+        add_item(title=task_title, status=task_status)
+        return redirect('/')
+    else:
+        all_tasks = get_items()
+        return render_template('index.html', tasks=all_tasks)
+
+#new task
+@app.route('/new', methods=['GET', 'POST'])
+def new_post():
+    if request.method == 'POST':
+        task_title = request.form['title']
+        task_status = request.form['status']
+        add_item(title=task_title, status=task_status)
+        return redirect('/')
+    else:
+        return render_template('new_task.html')
```

Comment:
It would be neater to split this into two functions, one with `@app.route('/new', methods=['GET'])` and one with `@app.route('/new', methods=['POST'])`, instead of doing your own routing with if/else